src/components/VDataTable/VDataTable.ts

```diff
@@ -14,14 +14,14 @@ import mixins from '../../util/mixins'
 import { VNode, CreateElement, VNodeData, VNodeChildren, VNodeChildrenArrayContents } from 'vue'
 import { PropValidator } from 'vue/types/options'
 
-export type TableHeader = {
+export interface TableHeader {
   text: string
   value: string
-  align: 'left' | 'center' | 'right'
-  sortable: boolean
-  class: string | string[]
-  width: string
-  filter: (v: any) => boolean
+  align?: 'left' | 'center' | 'right'
+  sortable?: boolean
+  class?: string | string[]
+  width?: string | number | null
```

Comment:
I meant the return value of computed, not here. I guess you should keep number though.